packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+---
+id: json-rpc-api
+title: JSON RPC Specification
+sidebar_label: JSON RPC Specification
+---
+
+In this section we provide an overview of the JSON RPC schema for the Nitro protocol. All methods accept requests and return responses following the [JSON RPC 2.0 specification](https://www.jsonrpc.org/specification).
+
+## CreateChannel
+
+This method is used to create a channel between a set of participants with given budgets. It contains information regarding type of channel and initial state of that channel.
+
+There are three different types of channels:
+
+- _Direct Channel_: channels that support applications between ledger channel counterparties
+- _Ledger Channel_: channels that manage the budgets and allocations between direct and virtual channels
+- _Virtual Channel_: channels that support applications routed over ledger channels
+
+In order to create a virtual or direct channel, you must have first created and funded a ledger channel between client and hub.
+
+A `Participant` contains identifying information about channel members:
+
+| Name           | Type      | Description                                              |
+| -------------- | --------- | -------------------------------------------------------- |
+| participantId  | string    | Unique identifier of channel participant                 |
+| signingAddress | `Address` | Address used to sign channel state updates               |
+| destination    | `Address` | Address to receive funds in case of disputes/withdrawals |
```

Comment:
`destination` is of type `Destination`

Reply:
It seems like it's an `Address` type in the `data-types` [file](https://github.com/connext/statechannels/blob/6dbfe6c96a9973fb8d1e48cac4a6e640d7bd7e99/packages/client-api-schema/src/data-types.ts#L37-L50):

```typescript
/**
 * Container for data specific to a single state channel participant
 */
export interface Participant {
  /**
   * App allocated id, used for relaying messages to the participant
   */
  participantId: string;
  /**
   * Address used to sign channel updates
   */
  signingAddress: Address;
  /**
   * Address of EOA to receive channel proceeds (the account that'll get the funds).
   */
  destination: Address;
}
```

What would the `Destination` type look like?

Reply:
It is a `bytes32` which may or may not be a converted left padded `address` (i.e., it is either a `ChannelId` or an `Address`)

Reply:
Little bit of extra context: We can (and do) add annotations to the declarations in that file, such that we can actually enforce stricter checks at run time. Example (left padded address): 

https://github.com/statechannels/statechannels/blob/c831b2f14ac086567989db3691a6a1f0adcb3298/packages/client-api-schema/src/data-types.ts#L14-L18

[Here ](https://docs.statechannels.org/docs/protocol-tutorial/deposit-assets#destinations)is the corresponding docs entry.

As you can see from the code comment, this work is a little incomplete.

I think in this case `destination` ought to remain an `Address`, since here we are sure we want an ethereum address. In other places, `Destination` is right.